packages/stream-model-instance/src/model-instance-document.ts

```diff
@@ -264,11 +298,26 @@ export class ModelInstanceDocument<T = Record<string, any>> extends Stream {
       prev: this.tip,
       id: this.id.cid,
     }
+    // Null check is necessary to avoid `undefined` value that can't be encoded with IPLD
+    if (metadata.shouldIndex != null) {
+      rawCommit.header = {
+        shouldIndex: metadata.shouldIndex,
+      }
+    }
     const commit = await signer.createDagJWS(rawCommit)
     const updated = await this.api.applyCommit(this.id, commit, opts)
     this.state$.next(updated.state)
   }
 
+  /**
+   * Set the index metadata field for the stream
+   * @param shouldIndex - Whether the stream should be indexed or not
+   * @param opts - Additional options
+   */
+  shouldIndex(shouldIndex: boolean, opts: UpdateOpts = {}): Promise<void> {
```

Comment:
I'd vote for calling this method `setShouldIndex`.  A method called `shouldIndex` sounds to me like it would be a read-only accessor method that returns a boolean.